src/craig_bampton.jl

```diff
@@ -0,0 +1,42 @@
+# This file is a part of JuliaFEM.
+# License is MIT: see https://github.com/JuliaFEM/ModelReduction.jl/blob/master/LICENSE
+
+"""
+    craig_bampton(K, M, r, l, n)
+
+Reduce the stiffness and mass matrices with the Craig-Bampton method.
+K = original stiffness matrix, M = original mass matrix,
+r = retained DOF:s, l = internal DOF:s, n = the number of modes to keep.
+
+"""
+function craig_bampton(K, M, r, l, n)
+    eps_ = 0.000001
+    Krr = K[r,r]; Krl = K[r,l]
+    Klr = K[l,r]; Kll = K[l,l]
+    Mrr = M[r,r]; Mrl = M[r,l]
+    Mlr = M[l,r]; Mll = M[l,l]
+    w2 = eigvals(Kll,Mll)
+    X1 = eigvecs(Kll,Mll)
+    X1[abs.(X1) .< eps_] = 0
+    X = X1[:,1:n]
+    V = X1'*Kll*X1
+    Kmm = X'*Kll*X
+    Kmm[abs.(Kmm) .< eps_] = 0
+    Kbb = Krr + Krl*(-X1*inv(V)*X1'*Klr)
```

Comment:
do invV = inv(V) and reuse that everywhere so you don't have to calculate it multiple times